src/plugins/intel_cpu/src/emitters/plugin/x64/jit_bf16_emitters.hpp

```diff
@@ -33,45 +33,63 @@ class jit_uni_vcvtneps2bf16 : public jit_emitter {
         }
     }
 
+    template <typename Vmm>
+    void saturate_input(Vmm& clamped,
+                        const Vmm& in,
+                        const std::string& bf16_min_key,
+                        const std::string& bf16_max_key) const {
+        Vmm vmm_temp = Vmm(aux_vec_idxs[1]);
+        h->uni_vmovups(vmm_temp, table_val(bf16_min_key));
+        h->uni_vmaxps(clamped, in, vmm_temp);
+        h->uni_vmovups(vmm_temp, table_val(bf16_max_key));
+        h->uni_vminps(clamped, clamped, vmm_temp);
+
+        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core)) {
+            h->uni_vmovups(vmm_temp, table_val("selector"));
+            h->vfixupimmps(clamped, in, vmm_temp, 0);
+        }
+    }
+
     template <dnnl::impl::cpu::x64::cpu_isa_t isa>
     void emit_isa(const std::vector<size_t>& in_vec_idxs, const std::vector<size_t>& out_vec_idxs) const {
         using namespace Xbyak;
         using Vmm = typename dnnl::impl::utils::conditional3<isa == dnnl::impl::cpu::x64::sse41, Xmm, isa == dnnl::impl::cpu::x64::avx2, Ymm, Zmm>::type;
 
         Vmm in = Vmm(in_vec_idxs[0]);
+        Vmm clamped = Vmm(aux_vec_idxs[0]);
+        saturate_input(clamped, in, "bf16_min", "bf16_max");
 
         if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_bf16)) {
             Ymm out = Ymm(out_vec_idxs[0]);
-            h->vcvtneps2bf16(out, in);
+            h->vcvtneps2bf16(out, clamped);
         } else if (host_isa_ == dnnl::impl::cpu::x64::cpu_isa_t::avx512_core) {
-            Zmm aux = Zmm(aux_vec_idxs[0]);
-            Zmm aux1 = Zmm(aux_vec_idxs[1]);
+            Zmm aux = Zmm(aux_vec_idxs[1]);
+            Zmm aux1 = Zmm(aux_vec_idxs[2]);
             Ymm out = Ymm(out_vec_idxs[0]);
 
-            h->uni_vpsrld(aux, in, 16);
+            h->uni_vpsrld(aux, clamped, 16);
             h->vpandd(aux, aux, table_val("one"));
             h->uni_vmovups(aux1, table_val("even"));
             h->uni_vpaddd(aux, aux1, aux);
-            h->uni_vpaddd(aux, in, aux);
-            h->vfixupimmps(aux, in, table_val("selector"), 0);
+            h->uni_vpaddd(aux, clamped, aux);
```

Comment:
Could you please double check we can remove` h->vfixupimmps(aux, in, table_val("selector"), 0);` here. Lines above have possibility to spoil the inf/nan bit again.

Reply:
when adding back h->vfixupimmps(aux, in or clamped, table_val("selector"), 0); here, the results seems goes wrong for avx512_core platforms.